contracts/ProfileToken.sol

```diff
@@ -52,30 +51,47 @@ contract ProfileToken is
     }
 
     // associate an avatar with a profile. must be owner of both tokens.
-    function setAvatar(address avatarContractAddrs, uint avatarTokenId, uint niftyTokenId) public {
-        require(msg.sender == ERC721.ownerOf(niftyTokenId), "not profile owner");
+    function setAvatar(
+        address avatarContractAddrs,
+        uint256 avatarTokenId,
+        uint256 niftyTokenId
+    ) public {
+        require(
+            msg.sender == ERC721.ownerOf(niftyTokenId),
+            "not profile owner"
+        );
 
         ERC721 avatarNFT = ERC721(avatarContractAddrs);
-        require(msg.sender == avatarNFT.ownerOf(avatarTokenId), "not avatar owner");
-
-        _avatarInfo.nftContract = avatarContractAddrs;
-        _avatarInfo.tokenId = avatarTokenId;
-        
-         // Get the TokenURI for the Avatar NFT
-        string memory AvatartokenURI = avatarNFT.tokenURI(avatarTokenId);
-
-        //  Update the NiftyTokenId to Avatar Map
-        _tokenURIs[niftyTokenId] = AvatartokenURI;
-
-        emit SetAvatar(niftyTokenId, avatarContractAddrs, avatarTokenId, AvatartokenURI);
+        require(
+            msg.sender == avatarNFT.ownerOf(avatarTokenId),
+            "not avatar owner"
+        );
+
+        _avatars[niftyTokenId].nftContract = avatarContractAddrs;
+        _avatars[niftyTokenId].tokenId = avatarTokenId;
+
+        // Get the TokenURI for the Avatar NFT
+        string memory avatartokenURI = avatarNFT.tokenURI(avatarTokenId);
+
+        emit SetAvatar(
+            niftyTokenId,
+            avatarContractAddrs,
+            avatarTokenId,
+            avatartokenURI
+        );
     }
 
     // pass through to avatar contract
-    function tokenURI(uint256 tokenId) public view virtual override(ERC721) returns (string memory) {
-
-        require(_exists(tokenId), "Querying for a non existent token");
-
-         return _tokenURIs[tokenId];
+    function tokenURI(uint256 tokenId)
+        public
+        view
+        virtual
+        override(ERC721)
+        returns (string memory)
+    {
+        require(_exists(tokenId), "non-existent token");
+        IERC721Metadata avatarNFT = IERC721Metadata(_avatars[tokenId].nftContract);

```

Comment:
@ewerx can you explain this line? Sorry if this is supposed to be something very obvious, but I'm not understanding it!

Reply:
added comment